packages/connection-storage/src/connection-storage.ts

```diff
@@ -229,6 +232,10 @@ export class ConnectionStorage {
       );
     }
     const connectionInfo = mergeSecrets(storedConnectionInfo!, secrets);
+    connectionInfo.userFavorite ??= !!connectionInfo.favorite;
```

Comment:
a silly ask, but i can totally see how this info will get lost in 6 months, can we take these 3 lines, put them in a `migrateFromOldFavorites()`, and add a short comment on top of it with some context / a link to a ticket or technical design?

Reply:
That makes a lot of sense. Will do! I'm still drafting a few changes though but I'm including this one.